packages/fuselage-ui-kit/src/elements/ChannelsSelectElement/ChannelsSelectElement.tsx

```diff
@@ -0,0 +1,83 @@
+import { AutoComplete, Option, Box, Options } from '@rocket.chat/fuselage';
+import { useDebouncedValue } from '@rocket.chat/fuselage-hooks';
+import { RoomAvatar } from '@rocket.chat/ui-avatar';
+import type * as UiKit from '@rocket.chat/ui-kit';
+import { memo, useCallback, useMemo, useState } from 'react';
+
+import { useUiKitState } from '../../hooks/useUiKitState';
+import type { BlockProps } from '../../utils/BlockProps';
+import { useChannelsData } from './hooks/useChannelsData';
+
+type ChannelsSelectElementProps = BlockProps<UiKit.ChannelsSelectElement>;
+
+const ChannelsSelectElement = ({
+  block,
+  context,
+}: ChannelsSelectElementProps) => {
+  const [{ value, loading }, action] = useUiKitState(block, context);
+
+  const [filter, setFilter] = useState('');
+  const filterDebounced = useDebouncedValue(filter, 300);
+
+  const data = useChannelsData({ filter: filterDebounced });
+
+  const options = useMemo(
```

Comment:
This should be either calculated in the useQuery hook fetch function, or passed as a funtion to `select` property of useQuery.